ui/src/export.tsx

```diff
@@ -720,178 +730,315 @@ function ExportDialog(
+        {stage === ExportDialogStage.EXPORTING ? (
+          <GridBox sx={{ minHeight: 100 }}>
+            <Loading
+              status={exportState}
+              showProgressOnMutate
+              disableReloadButton
+            />
+          </GridBox>
+        ) : null}
+        {stage === ExportDialogStage.LINKS ||
+        stage === ExportDialogStage.REDIRECT_ERRORS ? (
+          <GridLayout rows spacing={1} height="auto">
+            <Typography variant="body1">
+              {stage === ExportDialogStage.LINKS
+                ? "Click on the filenames to download if they do not start automatically"
+                : "There were errors exporting one or more files"}
+            </Typography>
+            {orderedLinks.map((list) => (
+              <GridLayout key={list[0].tags[0]} rows spacing={1} height="auto">
+                {list[0].tags[0] ? (
+                  <Typography variant="body1">{list[0].tags[0]}</Typography>
+                ) : null}
+                <GridLayout rows height="auto">
+                  {list.map((link) => renderLink(link, stage))}
                 </GridLayout>
-                <GridBox sx={{ minHeight: 200 }}>
-                  {exporting ? (
-                    <CenteredContent progress text="Exporting..." />
-                  ) : (
-                    output ?? (
-                      <CenteredContent text="Choose an export method and click 'Export'" />
-                    )
-                  )}
-                </GridBox>
               </GridLayout>
-            ) : null}
+            ))}
           </GridLayout>
-        </Loading>
+        ) : null}
+        {stage === ExportDialogStage.REDIRECT ? (
+          <GridBox sx={{ minHeight: 100 }}>
+            <CenteredContent progress text="Redirecting..." />
+          </GridBox>
+        ) : null}
       </DialogContent>
       <DialogActions>
         <Button
           onClick={() => {
             props.hide();
-            setExporting(false);
+            setStage(ExportDialogStage.MODEL_SELECT);
           }}
         >
-          Close
+          {stage === ExportDialogStage.MODEL_SELECT ? "Cancel" : "Close"}
         </Button>
+        {stage === ExportDialogStage.MODEL_SELECT ? (
+          <Button
+            variant="contained"
+            onClick={() => {
+              setStage(ExportDialogStage.EXPORTING);
+              exportState.trigger();
+            }}
+          >
+            Export dataset
+          </Button>
+        ) : null}
+        {stage === ExportDialogStage.REDIRECT_ERRORS ? (
+          <Button
+            variant="contained"
+            onClick={() => {
+              setStage(ExportDialogStage.REDIRECT);
+              if (exportState.data?.redirectURL) {
+                redirect(
+                  exportState.data.redirectURL,
+                  absoluteExportURL(params)
+                );
+              }
+            }}
+          >
+            Continue
+          </Button>
```

Comment:
Does this bit give the user a option to Continue even if there were errors?

Reply:
Yes, that's what was in the mocks. Do you think it shouldn't?

Reply:
I think it's fine, as long as we also show the errors. Was just making sure I was following.